### PR TITLE
[Unit Test] Unit Test forgot password

### PR DESCRIPTION
**Pull request by Tuanlvt**
Unit Test forgot password

### CODE REVIEW

File: app/src/test/java/com/framgia/wsm/screen/forgotpassword/ForgotPasswordPresenterTest.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    private ImmediateSchedulerProvider mSchedulerProvider;
+
+    @Before
+    public void setUp() throws Exception {
+        mPresenter.setViewModel(mViewModel);
+        mPresenter.setBaseSchedulerProvider(mSchedulerProvider);
+    }
+
+    @After
+    public void tearDown() throws Exception {
```
Comment (daolq3012) [inline app/src/test/java/com/framgia/wsm/screen/forgotpassword/ForgotPasswordPresenterTest.java]: pls remove this method

File: app/src/test/java/com/framgia/wsm/screen/forgotpassword/ForgotPasswordPresenterTest.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
+package com.framgia.wsm.screen.forgotpassword;
+
+import com.framgia.wsm.data.source.RequestRepository;
+import com.framgia.wsm.data.source.remote.api.response.BaseResponse;
+import com.framgia.wsm.utils.rx.ImmediateSchedulerProvider;
+import com.framgia.wsm.utils.validator.Validator;
+import io.reactivex.Single;
+import org.junit.After;
```
Comment (trainer-android-fr) [inline app/src/test/java/com/framgia/wsm/screen/forgotpassword/ForgotPasswordPresenterTest.java]: ERROR: Unused import: org.junit.After. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

Comment (trainer-android-fr) [inline app/src/test/java/com/framgia/wsm/screen/forgotpassword/ForgotPasswordPresenterTest.java]: WARNING: [UnusedImports] Avoid unused imports such as 'org.junit.After'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/test/java/com/framgia/wsm/screen/forgotpassword/ForgotPasswordPresenterTest.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        boolean actual = mValidator.validateAll(mViewModel);
+
+        Assert.assertEquals(expect, actual);
+    }
+
+    @Test
+    public void validateEmailInput_shouldCorrect_whenInputValidData() throws Exception {
+        //Give
+        String email = "abc@framgia.com.edev";
+        String msgEmty = "Email less than 6 characters";
```
Comment (tranminhtri911) [inline app/src/test/java/com/framgia/wsm/screen/forgotpassword/ForgotPasswordPresenterTest.java]: (tat) (kill)

File: app/src/test/java/com/framgia/wsm/screen/forgotpassword/ForgotPasswordPresenterTest.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    private ImmediateSchedulerProvider mSchedulerProvider;
+
+    @Before
+    public void setUp() throws Exception {
+        mPresenter.setViewModel(mViewModel);
+        mPresenter.setBaseSchedulerProvider(mSchedulerProvider);
+    }
+
+    @Test
+    public void sendEmail_WhenSendEmailToSever() throws Exception {
```
Comment (daolq3012) [inline app/src/test/java/com/framgia/wsm/screen/forgotpassword/ForgotPasswordPresenterTest.java]: ``sendEmail_WhenSendEmailToSever`` -> ``sendEmai_Should....._WhenSendEmailToSever``

File: app/src/test/java/com/framgia/wsm/screen/forgotpassword/ForgotPasswordPresenterTest.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    @Test
+    public void sendEmail_shouldSuccess_whenInputValidDataAndNetworkSuccess() throws Exception {
+        //When
+        when(mRepository.sendEmail(EMAIL)).thenReturn(
+                Single.just(new BaseResponse(Mockito.anyString())));
+
+        //Then
+        mPresenter.sendEmail(EMAIL);
+
+        Mockito.verify(mViewModel, Mockito.never()).onSendEmailError(null);
```
Comment (trainer-android-fr) [inline app/src/test/java/com/framgia/wsm/screen/forgotpassword/ForgotPasswordPresenterTest.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'Mockito.verify' due to existing static import 'org.mockito.Mockito.verify'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/test/java/com/framgia/wsm/screen/forgotpassword/ForgotPasswordPresenterTest.java [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+        Mockito.verify(mViewModel).onSendEmailSuccess();
```
Comment (trainer-android-fr) [inline app/src/test/java/com/framgia/wsm/screen/forgotpassword/ForgotPasswordPresenterTest.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'Mockito.verify' due to existing static import 'org.mockito.Mockito.verify'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/test/java/com/framgia/wsm/screen/forgotpassword/ForgotPasswordPresenterTest.java [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+        String msgNotEmty = "Must not Emty";
+        String msgError = "Invalid email";
+
+        //When
+        when(mValidator.validateValueNonEmpty(email)).thenReturn(msgNotEmty);
+        when(mValidator.validateEmailFormat(email)).thenReturn(msgError);
+
+        //Then
+        mPresenter.validateEmailInput(email);
+        Mockito.verify(mViewModel, Mockito.never()).onInputEmailError(msgError);
```
Comment (trainer-android-fr) [inline app/src/test/java/com/framgia/wsm/screen/forgotpassword/ForgotPasswordPresenterTest.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'Mockito.verify' due to existing static import 'org.mockito.Mockito.verify'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/test/java/com/framgia/wsm/screen/forgotpassword/ForgotPasswordPresenterTest.java [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+        String msgNotEmty = "Must not Emty";
+        String msgError = "Invalid email";
+
+        //When
+        when(mValidator.validateValueNonEmpty(email)).thenReturn(msgNotEmty);
+        when(mValidator.validateEmailFormat(email)).thenReturn(msgError);
+
+        //Then
+        mPresenter.validateEmailInput(email);
+        verify(mViewModel, Mockito.never()).onInputEmailError(msgError);
```
Comment (daolq3012) [inline app/src/test/java/com/framgia/wsm/screen/forgotpassword/ForgotPasswordPresenterTest.java]: pls change to ``verify(mViewModel).onInputEmailError("");``

File: app/src/test/java/com/framgia/wsm/screen/forgotpassword/ForgotPasswordPresenterTest.java [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+
+    @Test
+    public void validateDataInput_shouldReturnFalse_whenInputValidDataError() throws Exception {
+        //Give
+        boolean expect = false;
+        //When
+        when(mValidator.validateAll(mViewModel)).thenReturn(false);
+
+        //Then
+        boolean actual = mValidator.validateAll(mViewModel);
```
Comment (daolq3012) [inline app/src/test/java/com/framgia/wsm/screen/forgotpassword/ForgotPasswordPresenterTest.java]: pls change to ``presenter.validateDataInput(Mockito.anyString())``

**Pull request merged**